### DISCUSSION TITLE
Test Kyuubi custom event handler on Kyuubi=1.7.1 does not meet expectation， did I configure right?

### DISCUSSION THREAD

**Discussion by madeirak** [apache/kyuubi#6795] [Q&A]
I followed this https://kyuubi.readthedocs.io/en/v1.7.1-rc0/extensions/server/events.html
and also refered to this : https://github.com/zhaomin1423/example-custom-event-handler/blob/dev/pom.xml

I had configure this conf : kyuubi.backend.server.event.loggers=CUSTOM
but It seems doesnt work?

Do I still need to configure class name like custom AUTH？？？
![image](https://github.com/user-attachments/assets/cc04dee4-dd11-4808-881e-b49524b61e63)

**Comment by wForget** [comment]
The custom event handler is loaded by SPI. Do you create a services file? Like: https://github.com/zhaomin1423/example-custom-event-handler/blob/dev/src/main/resources/META-INF/services/org.apache.kyuubi.events.handler.CustomEventHandlerProvider

**Reply by madeirak** [reply]
Thx for reply, I ultimately chose to send it directly to Kafka by configuring Kyuubi conf.
btw, I want to know how to get engine type from KyuubiOperationEvent msg, or which API can I use to obtain the EngineType using KyuubiOperationEvent info？

![image](https://github.com/user-attachments/assets/375a232c-c204-426b-a68b-13d07eebfae6)

**Reply by wForget** [reply]
Maybe you should determine which engine the session uses in session event and which session the operation belongs to.

**Reply by wForget** [reply]
Get the value of `kyuubi.engine.type` from `conf` of session event
